sdk/core/System.ClientModel/src/Internal/Logging/ClientModelLogMessages.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace System.ClientModel.Internal;
+
+// The methods in this class should only ever be called from LoggingHandler
```

Comment:
What imposes this requirement?  If it is a hard requirement, is there a way to design the type to enforce it?

Reply:
This is required in order to provide the desired behavior of logging to event source if no ILogger is provided. The abstraction is meant to alleviate problems such as future devs forgetting to log to one or the other.

Reply:
Great.  It would be good if we could find an implementation that enforces that, so folks don't accidentally call these if they didn't read the comment.